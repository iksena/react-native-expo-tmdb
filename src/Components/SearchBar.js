import { Icon, Input } from 'native-base';
import { MaterialIcons } from '@expo/vector-icons';

function SearchBar({
  value, setValue, onClose, onSearch,
}) {
  return (
    <Input
      w="80%"
      value={value}
      onChangeText={setValue}
      InputRightElement={(
        <Icon
          as={(
            <MaterialIcons
              name={value !== '' ? 'close' : 'search'}
              onPress={value !== '' ? onClose : onSearch}
            />
            )}
          size={5}
          ml="2"
          color="muted.500"
        />
        )}
      placeholder="Search TV Shows"
      onSubmitEditing={onSearch}
      returnKeyType="search"
    />
  );
}

export default SearchBar;
