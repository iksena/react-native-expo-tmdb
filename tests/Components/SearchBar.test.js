import { render } from '../test-utils';
import SearchBar from '../../src/Components/SearchBar';

jest.mock('@expo/vector-icons', () => ({
  MaterialIcons: 'MaterialIcons',
}));

describe('SearchBar component', () => {
  let props = {};

  beforeEach(() => {
    props = {
      value: '',
      setValue(value) {
        this.value = value;
      },
      onClose: jest.fn(),
      onSearch: jest.fn(),
    };
  });

  describe('#render', () => {
    it('should render component', () => {
      const { toJSON } = render(<SearchBar {...props} />);

      expect(toJSON()).toMatchSnapshot();
    });

    it('should render component with close button when user search', () => {
      props.value = 'search input';

      const { toJSON } = render(<SearchBar {...props} />);

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
