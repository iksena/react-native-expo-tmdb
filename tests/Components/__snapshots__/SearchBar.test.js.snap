// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component #render should render component 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#e5e5e5",
        "borderRadius": 4,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "overflow": "hidden",
        "width": "80%",
      }
    }
  >
    <TextInput
      accessible={true}
      allowFontScaling={true}
      dataSet={Object {}}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onSubmitEditing={[MockFunction]}
      placeholder="Search TV Shows"
      placeholderTextColor="#a3a3a3"
      rejectResponderTermination={true}
      returnKeyType="search"
      secureTextEntry={false}
      selectionColor="#1f2937"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#1f2937",
            "flex": 1,
            "fontFamily": undefined,
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": undefined,
          },
          Object {},
        ]
      }
      underlineColorAndroid="transparent"
      value=""
      variant="unstyled"
    />
    <MaterialIcons
      dataSet={Object {}}
      name="search"
      onPress={[MockFunction]}
      style={
        Object {
          "color": "#737373",
          "fontSize": 20,
          "height": 20,
          "lineHeight": 20,
          "marginLeft": 8,
          "width": 20,
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;

exports[`SearchBar component #render should render component with close button when user search 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#e5e5e5",
        "borderRadius": 4,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "overflow": "hidden",
        "width": "80%",
      }
    }
  >
    <TextInput
      accessible={true}
      allowFontScaling={true}
      dataSet={Object {}}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onSubmitEditing={[MockFunction]}
      placeholder="Search TV Shows"
      placeholderTextColor="#a3a3a3"
      rejectResponderTermination={true}
      returnKeyType="search"
      secureTextEntry={false}
      selectionColor="#1f2937"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#1f2937",
            "flex": 1,
            "fontFamily": undefined,
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": undefined,
          },
          Object {},
        ]
      }
      underlineColorAndroid="transparent"
      value="search input"
      variant="unstyled"
    />
    <MaterialIcons
      dataSet={Object {}}
      name="close"
      onPress={[MockFunction]}
      style={
        Object {
          "color": "#737373",
          "fontSize": 20,
          "height": 20,
          "lineHeight": 20,
          "marginLeft": 8,
          "width": 20,
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
